src/lib/deploy.js

```diff
@@ -451,6 +476,17 @@ async function deploy({ alias, yes }) {
   await shutdown();
 }
 
+/*
+While SnarkyJS and the zkApp CLI have a major version of 0,
+a change of the minor version represents a breaking change.
+When SnarkyJS and the zkApp have a major version of 1 or higher,
+changes to the major version of the zkapp-cli will represnt
+breaking changes, following semver.
+*/
+function hasBreakingChanges(version1, version2) {
+  return version1.split('.')[1] !== version2.split('.')[1];
```

Comment:
Would encourage we add the logic to handle this whether the major version is 0 or 1. Easy to do and ensures we don't overlook it when that day comes